frame/staking/src/mock.rs

```diff
@@ -533,7 +534,7 @@ fn post_conditions() {
 }
 
 fn check_count() {
-	let nominator_count = Nominators::<Test>::iter().count() as u32;
+	let nominator_count = Nominators::<Test>::iter_keys().count() as u32;
```

Comment:
the reason for this is subtle and rather tricky.

Reply:
Because we can fail to decode?

Reply:
yes we might fail to decode the _values_ (in the rare case that we suddenly reduce `MaxNominators` to e.g. 4), so `iter().count()` would return only the number of decodable ones, while `iter_keys()` is all of the nominators.